### DISCUSSION TITLE
Idiot's guide to ordering a PCB?

### DISCUSSION THREAD

**Discussion by tobymurray** [vroland/epdiy#354] [Q&A] [answered]
I have never ordered a PCB before. I'm open to trying it, but my skill are almost exclusively software so the hardware is well beyond me.

Has anyone made an Idiot's Guide to ordering this? I appreciate that 99% of the work has been done and it should largely be as simple as "upload and pay" but e.g. uploading https://vroland.github.io/epdiy-hardware/build/boards/epdiy-v7_top_pos.csv and https://vroland.github.io/epdiy-hardware/build/boards/epdiy-v7_BoM.csv to PCBWay yielded "Audit failed":
![image](https://github.com/user-attachments/assets/8e7ebfb4-0189-4a88-ba72-60ed8c22597e)

![image](https://github.com/user-attachments/assets/3e22e605-2cff-4a11-a676-2be079198472)

As far as I know it's a 2-layer PCB, but I don't know how to reupload the file in such a way as to express that.

Is trying to buy via this random person's "project" on PCBWay a sensible way to do it? https://www.pcbway.com/project/shareproject/epdiy_v7_E_Paper_driver_board_90c56cd0.html

**Comment by martinberlin** [comment]
Hello Toby,

> As far as I know it's a 2-layer PCB

It is a 4 layer PCB. Try telling PCBway that is 4 and check if that fixes the issue

**Reply by tobymurray** [reply]
Obviously I don't know very far! Thanks, I'll give it a shot.

**Reply by tobymurray** [reply]
Now I'm getting a prompt that claims it is optional but is decidedly not:

<img src="https://github.com/user-attachments/assets/0c274ea2-3860-4bee-a883-fb7c98b5e5c8" width="120">

Taking a shot in the dark from the image above, would it be "TL", "IN1", "IN2", "BL"?

**Reply by martinberlin** [reply]
I guess you can leave it blank. But if you are unsure:
![Layer order](https://github.com/user-attachments/assets/dfccaeda-36b5-44c9-bcff-cd5fcee7d601)

In my opinion, you don't need to know about any IDE program like KiCad or Altium to send something to fabrication. But getting to know the basics of them, makes you feel a bit more sure of what you are doing, when sending something to fabrication.

So I highly recommend it. Just download KiCad and check the basics, you don't need to design, you can just preview the PCB and check how many layers it has, etc. That gives you also confidence that what you are sending is what you see. 
For example: There was some Questions about some J* that where in the BOM but not in the PCB components, and they just stop there. Well if you know how to check the Schematics or even just the PCB, you can see that those J* are either Jumpers or just header holes, that are only needed in Gerbers. But as they "could be populated too" KiCad exports them in the BOM and Pick&Place file (unless you tell the software not to do it explicitly)
Also if you would like to try epdiy v7 before sending a full batch to fabrication there is still one available in Tindie, just search for "epdiy", since I don't want to self-advertise the Hardware here and I also like that other people like you has the initiative and courage to send PCBs to fabrication

**Reply by tomfleet** [reply]
Is there a reason as to why the non placed parts are not being excluded from the pnp data/BOM?

If full output is being provided, could it be considered reasonable that perhaps the production outputs should be generated for the "standard" config of dnp/dnl?

It is a fair comment, that if something is not to be placed by machine, or at all, that those lines probably shouldn't be in that specific production data release.

**Reply by martinberlin** [reply]
> Is there a reason as to why the non placed parts are not being excluded from the pnp data/BOM?

Not really. Maybe we just forgot to exclude it...
Yes sure. It should be excluded in KiCAD PCB and that should get it out of the .csv bom. 
I will make a PR in the next days to do that. Apart of that it would be nice that the **supported displays** table says more explicitly what version of the PCB has the right FPC connectors. 
This will make the selection of the PCB easier. For me at the moment the best selection is always v7 that uses S3. But that depends on everyone personal needs for a project.

**Comment by tobymurray** [comment]
For posterity, continuing on through PCBWay yields a (hopefully) landed cost of 201$ USD for the minimum quantity (5). This works out to ~ 55$ CAD to the door per assembled board, if I'm understanding correctly. 
![image](https://github.com/user-attachments/assets/08ab1fbb-5b2b-432c-abcd-0750a4277994)

**Reply by martinberlin** [reply]
Yes that's more or less correct depending on the provider. You can only lower this using JLCPCB and ordering some of the parts in larger quantities, that is not your intention here. In JLC you can also use PCBA in just 2 PCB`s of a minimum of 5 (You will get costs lower, but also 3 unpopulated PCBs)
I've the last epdiy v7 unit in Tindie for this summer if you want to try it and evaluate if it works for your project before getting 5 assembled. Just take a look searching for "epdiy"

**Reply by tobymurray** [reply]
Looks like my order has been assembled, PCBWay asked:

> Pls kindly check the pictures for  assembly and pay attention to the orientation of components, thanks. 

Specifically, they appear to be wondering the LED orientation:
![image](https://github.com/user-attachments/assets/55f2065f-2478-4c12-a39b-056e89bfd5da)

I would _guess_ that this means K is for "cathode" and A is for "anode". Does this seem right? I opened up KiCad but I'm afraid it has gotten me no further:
![image](https://github.com/user-attachments/assets/5dadc589-4813-4272-9beb-b688b954bd15)

Which way is which? Is the line part on the silkscreen the cathode, and PCBWay has it correct?

**Reply by martinberlin** [reply, marked as answer]
The Cathode or negative side should be facing the Resistor.
The current flows from Anode, to Cathode, resistor and then the TP4056. If it's wrong in the worst case it won't be bright but is just an indicator LED so it's not such a bad error.
![TP4056](https://github.com/user-attachments/assets/cc367082-caa4-4266-b484-b115f4631b2a)

![images](https://github.com/user-attachments/assets/1b1220b8-5491-4e1e-8687-8275b9770e4b)

**Reply by tobymurray** [reply]
>  (Opposite than the reference above)

What is the reference you're referring to here? As far as I can tell, the cathode appears to be on the resistor side in both the image PCBWay sent me and in the KiCad.

> If it's wrong in the worst case it won't be bright but is just an indicator LED so it's not such a bad error.

Reassuring! I told PCBWay it was correct.

**Reply by martinberlin** [reply]
> Reassuring! I told PCBWay it was correct.

Yes it is. Probably I look wrong the image as you can see the cathode has the cross part and the positive side is on the PCB edge.

Toby I'm trying to reply in the right direction. You need to actually check what you send to fabrication yourself, it can be that many times, even exporting everything correctly from KiCad an element is bad placed. That's why I recommend everyone to learn at least the KiCad basics to open a design and check it out for yourself.
If that is not OK for you, then you will be better off, buying a tested and working controller. This my opinion and I do not pretend that everyone agrees with it

**Comment by Ivy-Mo** [comment]
Sometimes I want to order and find that I can't select many specifications, which ends up causing me to waste a lot of time without getting the results I want, if you can provide Gerber, I can help you get a quote
